### PR TITLE
Add CFITSIO to recipe for fits files 

### PR DESCRIPTION
**Pull request by csedwar**
cfitsio is in conda-forge added to meta.yaml, modified build.sh and set_bld_opts.bat to include what i think are the right options (more uncertain about windows options)

Checklist
* [X] Used a fork of the feedstock to propose changes
* [X] Bumped the build number (if the version is unchanged)
* [X] Ensured the license file is being packaged.

### CODE REVIEW

File: recipe/build.sh [hunk 1]
```diff
@@ -33,6 +33,7 @@ bash configure --prefix=$PREFIX \
                --with-geos=$PREFIX/bin/geos-config \
                --with-geotiff=$PREFIX \
                --with-hdf4=$PREFIX \
+               --with-cfitsio=$PREFIX \
```
Comment (ocefpaf) [inline recipe/build.sh]: Can you add a test for it?

Comment (gillins) [reply recipe/build.sh]: > Can you add a test for it?

Just to clarify: In `run_test.py` from line 11.

Comment (csedwar) [reply recipe/build.sh]: Should be done.  Had a typo "FTIS" vs "FITS" but I expect to pass.  Let me know what else you need

**Comment by ocefpaf** [comment]
Looks like we have `cfitsio` on Windows, any reason to not add it here?

**Comment by csedwar** [comment]
I thought it would be straightforward too.

I tried to compile a few times but could not figure out the appropriate configure options. Any suggestions?

**Comment by ocefpaf** [comment]
> I thought it would be straightforward too. I tried to compile a few times but could not figure out the appropriate configure options. Any suggestions?

I would download the `cfitsio` Windows package, check where the `.lib` file lives, and then add it to `set_bld_opts.bat`. (You'll probably have to check the cmake options for `cfitsio`).

If that is too difficult you will have to skip the test on Windows so we can merge this.

**Comment by csedwar** [comment]
Will try a few more things and see if I can get it to compile. 

I have some but not much windows experience so will see what I can do.

**Comment by gillins** [comment]
Looks like adding the following to `set_bld_opts.bat` should work (some [comments](https://github.com/OSGeo/gdal/blob/master/gdal/nmake.opt#L534-L537) in `nmake.opt`):
```
FITS_PLUGIN=NO ^
FITS_INC_DIR=%LIBRARY_INC% ^
FITS_LIB="%LIBRARY_LIB%\cfitsio.lib" ^
```

File: recipe/set_bld_opts.bat [hunk 2]
```diff
@@ -32,6 +32,9 @@ set BLD_OPTS=%WIN64% ^
     EXPAT_DIR=%LIBRARY_PREFIX% ^
     EXPAT_INCLUDE="-I%LIBRARY_INC%" ^
     EXPAT_LIB=%LIBRARY_LIB%\expat.lib ^
+		FITS_PLUGIN=NO ^
+		FITS_INC_DIR=%LIBRARY_INC% ^
+		FITS_LIB="%LIBRARY_LIB%\cfitsio.lib ^
```
Comment (gillins) [inline recipe/set_bld_opts.bat]: Missing quote at end of this line (before the `^`). Also fix the indentation.

**Comment by csedwar** [comment]
Sigh. Fixed.

**Comment by gillins** [comment]
Thanks @csedwar !

**Pull request merged**